### PR TITLE
SelectControl: Fix remaining 40px size violations

### PR DESCRIPTION
**Pull request by mirka**
Part of #63871

## What?

Fixes the remaining 40px size violations for SelectControl.

## Testing Instructions

See inline comments.

### CODE REVIEW

File: packages/block-library/src/post-featured-image/dimension-controls.js [hunk 1]
```diff
@@ -236,8 +236,7 @@ const DimensionControls = ( {
 					panelId={ clientId }
 				>
 					<SelectControl
-						// TODO: Switch to `true` (40px size) if possible
-						__next40pxDefaultSize={ false }
+						__next40pxDefaultSize
```
Comment (mirka) [inline packages/block-library/src/post-featured-image/dimension-controls.js:239]: | Before | After |
|--------|-------|
|Featured Image block inspector, before|Featured Image block inspector, after|

File: packages/widgets/src/blocks/legacy-widget/edit/widget-type-selector.js [hunk 2]
```diff
@@ -27,8 +27,7 @@ export default function WidgetTypeSelector( { selectedId, onSelect } ) {
 
 	return (
 		<SelectControl
-			// TODO: Switch to `true` (40px size) if possible
-			__next40pxDefaultSize={ false }
+			__next40pxDefaultSize
```
Comment (mirka) [inline packages/widgets/src/blocks/legacy-widget/edit/widget-type-selector.js:30]: Activate a legacy theme like Twenty Twelve. Go to Appearance ▸ Widgets, and insert a Legacy Widget block.

| Before | After |
|--------|-------|
|Legacy Widget block placeholder, before|Legacy Widget block placeholder, after|

**Review by ciampo — APPROVED** [review]
LGTM 🚀 

I may have gotten a bit lost on tasks dependencies, but what prevented us from enabling `__next40pxDefaultSize` before on those instances? Just to make sure I have full context.

**Comment by mirka** [comment]
> I may have gotten a bit lost on tasks dependencies, but what prevented us from enabling `__next40pxDefaultSize` before on those instances? Just to make sure I have full context.

Good question. There were a lot more [TODO'ed violations](https://github.com/WordPress/gutenberg/pull/64486/files) when I first added the lint rule, but by now it had already whittled down to two remaining violations! Mostly because I was working on upsizing other components, and if there was a SelectControl in the same screen as those I would upsize it at the same time.

**Pull request merged**